src/main/java/com/rackspace/salus/telemetry/ambassador/services/AgentHistoryService.java

```diff
@@ -77,14 +82,14 @@ public AgentHistory addAgentHistory(EnvoySummary request,
 
   public void addEnvoyConnectionClosedTime(String tenantId, String envoyId)  {
     Optional<AgentHistory> agent = agentHistoryRepository.findByTenantIdAndEnvoyId(tenantId, envoyId);
-    if(!agent.isEmpty()) {
+    if(!agent.isPresent()) {
```

Comment:
It seems like this logic is flipped. If switching to `isPresent` I think it should be
```suggestion
    if(agent.isPresent()) {
```

A unit test should have caught this.

Reply:
have to write a test case for that.